tests/ut/runtime/test_execute_artifact.py

```
@@ -0,0 +1,119 @@
+def test_execute_artifact_dir_wires_compile_then_execute(tmp_path):
+    chip = object()
+    with (
+        patch(
+            "pypto.runtime.device_runner.compile_and_assemble",
+            return_value=(chip, "tensormap_and_ringbuffer", {}),
+        ) as comp,
+        patch.object(execute_artifact, "_execute_on_device") as exec_on_dev,
+    ):
+        execute_artifact_dir(tmp_path, "a2a3", 1, pto_isa_commit="deadbeef")
+    comp.assert_called_once_with(tmp_path, "a2a3", pto_isa_commit="deadbeef")
+    args, kwargs = exec_on_dev.call_args
```

Ruff (0.15.18)

[warning] Unpacked variable `kwargs` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)